api/datasets.go

```diff
@@ -73,4 +78,35 @@ func ConfigureDatasetsRouter(e *echo.Group, dldm *core.DeltaDM) {
 
 		return c.JSON(200, ads)
 	})
+
+	datasets.PUT("/:dataset_id", func(c echo.Context) error {
+		did := c.Param("dataset_id")
+		if did == "" {
+			return fmt.Errorf("dataset id not specified")
+		}
+
+		var d DatasetPutBody
+
+		if err := c.Bind(&d); err != nil {
+			return err
+		}
+
+		var existing db.Dataset
+		res := dldm.DB.Model(&db.Dataset{}).Where("id = ?", did).First(&existing)
+
+		if res.Error != nil {
```

Comment:
just confirming - will this also catch an issue where the `id` does not exist in the db?

If so, could you add a custom error message for that so the user sees ex) `Dataset ID 7 does not exist`

Reply:
this is a generic error catch. I change the error message to include the dataset id `fmt.Errorf("error fetching dataset %s: %s", did, res.Error)`. But on fetching dataset id that does not exist it will return `error fetching dataset 12313: record not found`